src/Core/Resolvers/CosmosQueryEngine.cs

```diff
@@ -137,16 +139,16 @@ public class CosmosQueryEngine : IQueryEngine
         /// Executes the given IMiddlewareContext of the GraphQL query and
         /// expecting a list of Json back.
         /// </summary>
-        public async Task<Tuple<IEnumerable<JsonDocument>, IMetadata>> ExecuteListAsync(IMiddlewareContext context, IDictionary<string, object> parameters)
+        public async Task<Tuple<IEnumerable<JsonDocument>, IMetadata>> ExecuteListAsync(IMiddlewareContext context, IDictionary<string, object> parameters, string dataSourceName = "")
         {
             // TODO: fixme we have multiple rounds of serialization/deserialization JsomDocument/JObject
             // TODO: add support for nesting
             // TODO: add support for join query against another container
             // TODO: add support for TOP and Order-by push-down
 
             CosmosQueryStructure structure = new(context, parameters, _metadataStoreProvider, _authorizationResolver, _gQLFilterParser);
-
-            Container container = _clientProvider.Client.GetDatabase(structure.Database).GetContainer(structure.Container);
+            CosmosClient client = string.IsNullOrEmpty(dataSourceName) ? _clientProvider.Clients[_clientProvider._defaultDataSourceName] : _clientProvider.Clients[dataSourceName];
```

Comment:
I'm not too familiar with CosmosClient SDK, but making copies of Singleton CosmosClient objects seems like it could be problematic. At the end of this method, the CosmosClient object is disposed and I'm not sure if that would have negative consequences. (CosmosClient might not be disposed here, you should use the `using` keyword here regardless)

Reply:
Umm, we are making multiple copies here right? If there are multiple db's we will register multiple unique instances which have their own connection strings.